openwisp_users/api/mixins.py

```diff
@@ -212,6 +214,90 @@ class FilterSerializerByOrgOwned(FilterSerializerByOrganization):
     _user_attr = 'organizations_owned'
 
 
+class QuerySetRequestMixin(BaseQuerySetRequestMixin):
+    def get_queryset(self, request):
+        user = request.user
+        queryset = super().get_queryset(request)
+        # superuser can see everything
+        if user.is_superuser or user.is_anonymous:
+            return queryset
+        # non superusers can see only items
+        # of organizations they're related to
+        organization_filter = getattr(user, self._user_attr)
+        # if field_name organization then just organization_filter
+        if self.field_name == 'organization':
+            return queryset.filter(pk__in=organization_filter)
+        # for field_name other than organization
+        conditions = Q(**{'organization__in': organization_filter})
+        return queryset.filter(conditions)
+
+    def __init__(self, *args, **kwargs):
+        self._user_attr = kwargs.get('user_attr')
+        kwargs.pop('user_attr')
+        super().__init__(*args, **kwargs)
+
+
+class DjangoOrganizationFilter(filters.ModelChoiceFilter, QuerySetRequestMixin):
+    pass
+
+
+class DjangoOrganizationM2MFilter(
+    filters.ModelMultipleChoiceFilter, QuerySetRequestMixin
+):
+    pass
+
+
+class FilterDjangoOrganization(filters.FilterSet):
+    """
+    A custom filter set class that applies DjangoOrganizationFilter
+    to all ModelChoiceFilter & ModelMultipleChoiceFilterfilters.
+    """
+
+    @classmethod
+    def filter_for_field(cls, field, name, lookup_expr='exact'):
+        if isinstance(field, ForeignKey) and field.name != 'user':
+            return DjangoOrganizationFilter(
+                queryset=field.remote_field.model.objects.all(),
+                label=field.verbose_name.capitalize(),
+                field_name=field.name,
+                user_attr=cls._user_attr,
+            )
+        if (
+            isinstance(field, ManyToManyField) and field.name != 'user'
+        ):  # pragma: no cover
```

Comment:
Red flag :red_square: 

We have a test project where you can add an m2m relationship and write a test to cover this: https://github.com/openwisp/openwisp-users/blob/master/tests/testapp/models.py

Reply:
While generating test coverage locally, I accidentally left this comment in the code and forgot to remove it. My bad, and thank you for bringing it to my attention 👍